easybuild/tools/downloadtools.py

```diff
@@ -0,0 +1,187 @@
+        return path
+    else:
+        _log.warning("Download of %s to %s failed, done trying" % (url, path))
+        return None
+
+
+def download_svn(revision, url, path, timeout):
+    """
+    Download a svn repository and create a tarball out of it in the given path
+    timeout is ignored here
+    """
+    # transform url into something SvnRepository understands
+    if url.startswith('svn+'):
+        url = url[4:]
+    # revision could already be a part of the url, we don't want that
+    if url.endswith(str(revision)):
+        url = url[:-len(revision)]
+    # same for path
+    if path.endswith(str(revision)):
+        path = path[:-len(revision)]
+    path = os.path.join(path, 'repo')
+    # svn revisions are digit's so only get these, some packages have version='r1234'
+    int_revision = ''.join([x for x in revision if x.isdigit()])
+
+    try:
+        svnrepo = SvnRepository(url)
+    except AttributeError, err:
+        _log.debug('error checkout out svn repo %s', err)
+        raise err
+    svnrepo.export(path, int_revision)
+    # make a tarfile in the directory next to the repo
+    tarpath = os.path.join(path, '..', revision)
+    make_tarfile(tarpath, path)
+    # path will always have '/repo' in it, so the rm should be file
+    shutil.rmtree(os.path.join(path))
+    return tarpath
+
+
+def download_local_file(filename, url, path, timeout):
+    """
+    Dowloading a local file is just copying it
+    timeout is ignored here.
+    """
+    if not url.endswith(filename):
+        url = os.path.join(filename)
+    if url.startswith('file://'):
+        url = url[7:]
+    _log.debug("copying %s to %s", url, path)
+    try:
+        shutil.copy2(url, path)
+    except IOError, error:
+        if error.errno == 2:  # no such file or directory
+            return None
+        raise
+    return path
+
+
+def download_file(filename, url, path):
+    """
+    Download a file from the given url, to the specified path.
+    This function will parse the url and try to use the protcol that best matches the url to download it
+    if the protocol is a versioning control system it will do a checkout of the repo and use filename as the
+    revision to check out instead of the file.
+    it will save this to the path as a tarball
+    returns the name of the downloaded file/tarball
+    """
+    _log.debug("Trying to download %s from %s to %s", filename, url, path)
+
+    timeout = build_option('download_timeout')
+    if timeout is None:
+        # default to 10sec timeout if none was specified
+        # default system timeout (used is nothing is specified) may be infinite (?)
+        timeout = 10
+    _log.debug("Using timeout of %s seconds for initiating download" % timeout)
+
+    # make sure directory exists
+    basedir = os.path.dirname(path)
+    mkdir(basedir, parents=True)
+
+    protocol = url.split(':')[0]
+    if protocol not in PROTOCOL_MAP:
```

Comment:
if you replace this with the below, you can use the constants for the protocols in `PROTOCOL_MAP`:

```python
if protocol + PROTOCOL_SUFFIX not in PROTOCAL_MAP:
```